osh/builtin_meta.py

```diff
@@ -39,18 +39,30 @@
 
 
 class Eval(vm._Builtin):
-    def __init__(self, parse_ctx, exec_opts, cmd_ev, tracer, errfmt):
-        # type: (ParseContext, optview.Exec, CommandEvaluator, dev.Tracer, ui.ErrorFormatter) -> None
+
+    def __init__(self, parse_ctx, exec_opts, cmd_ev, expr_ev, tracer, errfmt):
+        # type: (ParseContext, optview.Exec, CommandEvaluator, expr_eval.ExprEvaluator, dev.Tracer, ui.ErrorFormatter) -> None
         self.parse_ctx = parse_ctx
         self.arena = parse_ctx.arena
         self.exec_opts = exec_opts
         self.cmd_ev = cmd_ev
+        self.expr_ev = expr_ev
         self.tracer = tracer
         self.errfmt = errfmt
 
     def Run(self, cmd_val):
         # type: (cmd_value.Argv) -> int
 
+        if cmd_val.typed_args:
+            t = typed_args.ReaderFromArgv(cmd_val.argv[1:], cmd_val.typed_args,
+                                          self.expr_ev)
+
+            block = t.PosCommand()
+            t.Done()
+
+            unused, status = self.cmd_ev.EvalBlock(block)
```

Comment:
is status unused here?    

I guess it's worth considering how to propgate status

```
$ eval 'false'; echo $?
1
```